### ISSUE TITLE
Retention policy scope for delete commands

### ISSUE THREAD

**Issue by kostasb** [influxdata/influxdb#8088]
**Feature request**: Support retention policy scope for series deletion commands such as `DROP MEASUREMENT` , `DROP SERIES` , `DELETE FROM`.

**Current behavior**: Delete commands have database-wide scope. As a result, it is not possible to delete a measurement under a specific retention policy without having it deleted from all retention policies in the database it lives in.

Consider:
```
db1.rp1.measurement
db2.rp2.measurement
```
There is currently no way to drop the measurement or some series of it, from a specific rp.

**Comment by jwyse** [comment]
+1

**Comment by ryantheleach** [comment]
Oh man.

This thoroughly confused me when attempting to delete data. (on 1.3, due to supported libraries on .net)

1. Chronograf fails to parse 

```
DELETE FROM /.*/ WHERE "time" >= '2018-02-28T02:26:08.0000000Z' AND "time" <= '2018-02-28T02:27:08.0000000Z'
```

`database not found`

```
DELETE FROM "SensorData"../.*/ WHERE "time" >= '2018-02-28T02:26:08.0000000Z' AND "time" <= '2018-02-28T02:27:08.0000000Z'
```

`received status code 400 from server: err: error parsing query: database not supported at line 1, char 1`

When it works fine from command line as no database was specified, yet if you specific a database, it fails.

Additionally this prevents me from having a slightly more flexible schema.
https://stackoverflow.com/questions/49022407/how-can-i-delete-measurements-within-a-time-range-for-a-given-rp

**Comment by EugenNY** [comment]
this missing feature makes the use of retention policies and CONTINUOUS QUERIES hardly usable...

**Comment by fanjun1980** [comment]
+1

**Comment by alespour** [comment]
+1

**Comment by nicolas17** [comment]
I created a few RPs to store aggregated data, created CQs to put new data in them, and backfilled them with some SELECT..INTO queries. Now I want to delete the old high-resolution data from the default RP.

- If I use `DELETE FROM telegraf.autogen.cpu WHERE time < now() - 30d`, I get "`ERR: error parsing query: retention policy not supported`" (ie. this bug).
- If I use `DELETE FROM cpu WHERE time < now() - 30d`, I would delete old `cpu` data for all RPs, including what I just aggregated.
- If I alter the `autogen` RP to retain data for 30 days, I would delete old data for all measurements in the `autogen` RP rather than only `cpu`.

Looks like the only solution is to do the aggregated backfilling for _all_ my measurements and only alter the RP when I'm completely done... for which I'll have to give my server a bigger disk first :sparkles:

**Comment by lansalot** [comment]
This is a pretty serious hole - any ETA for implementation?

**Comment by akamac** [comment]
Any roadmap for this feature?

**Comment by xPaw** [comment]
This is silly.

I did `use database.autogen` which said it switched to the database using a retention policy, and then `drop measurement name` which ended up deleting data from `autogen` AND another retention policy I had.

**Comment by akamac** [comment]
This needs to be reopened and addressed.

**Comment by hanynowsky** [comment]
Oh My goodness.
I cam across this issue to day when I had to delete some data from the default retention and the from another downgraded retention.
And all I get is:
``` ERR: error parsing query: retention policy not supported at line 1, char 1```

This is a high priority issue.

What is the point of DELETE if it cannot be applied to retention policies.

**Comment by andydavidson** [comment]
'wontfix'ing this is sinful.

**Comment by James00001** [comment]
Yes, clearly "stale bot" closed this one in error, someone with the authority to do so needs to mark this as re-opened.

**Comment by foobar** [comment]
still needed

**Comment by dgnorton** [comment]
Re-opened. This issue was closed automatically for being stale but is a valid issue.

**Comment by stevenaldinger** [comment]
this is insane. data was written to a 360d retention policy accidentally. is there seriously no way to fix this? we're just stuck with ~15 minutes of data for all our users for 360d? redis timeseries is looking like its worth investigating at this point between this and other influxDB-isms...

**Comment by teichhei** [comment]
Same issue. 3 years ongoing, seriously?
I can do a select * from "oneyear"."HVAC" but no way to get rid of the records.

**Comment by axeede** [comment]
same problem today- not possible ?

**Comment by andriy-pankiv-lemberg** [comment]
Same issue, 3 years for basic delete feature)
So the only way to do it is to:

1. move data to another table with default infinite RP
2. drop original table
3. delete needed data in this tmp table with default RP
4. move data back to new table with the same name and RP

It is ugly, but al least it will do the job

**Comment by kszarlej** [comment]
:+1:

**Comment by SadSadko** [comment]
so over 4 years, no comments, no ETA. if the problem is not regarding Flux QL it's not getting attention?
**This is ridiculous**

**Comment by LeonChadwick** [comment]
+1 also need this feature.  Without it we have to get things perfect first time around, every time, which is unrealistic.

**Comment by sfawcett123** [comment]
+1 This is needed

**Comment by merrilmathew** [comment]
+1

**Comment by mrwogu** [comment]
Still waiting for it

**Comment by Mac-lucky** [comment]
Still waiting

**Comment by barrettathome** [comment]
+1

**Comment by ascha191** [comment]
+1

**Comment by firesgc** [comment]
+1

**Comment by ferengi82** [comment]
+1

**Comment by alkissack** [comment]
+1

**Comment by tubemeister** [comment]
+1

Let me just add that I'm baffled this isn't implemented. Who on earth thought that was a good idea, this is almost as out there as the "weeks start on thursday because unix epoch was thursday" thing, another InfluxDB classic headscratcher. Whiskey Tango Foxtrot, Over?

**Comment by Muqeet1** [comment]
+1

**Comment by rayquaza912** [comment]
+1

**Comment by osulivanDE** [comment]
+1

**Comment by d123456temp** [comment]
+!

**Comment by Sesshoumaru-sama** [comment]
+1

**Comment by bcutter** [comment]
I simply can't delete any data.

Using `delete from "%" where "entity_id"='entity_id=h5055_batt';` seems to work (no output, so it seems to fail silentsly), but doing a select shows/confirms: data still exists.

Retention policy is open end (Duration: `∞`).

Using `delete from "homeassistant"."autogen"."%" where "entity_id"='entity_id=h5055_batt';` gives

`ERR: error parsing query: retention policy not supported at line 1, char 1`

⁉️⁉️⁉️ **HOW CAN I FREAKING GET RID OF DATA (single series)** ⁉️⁉️⁉️

**Comment by devanbenz** [comment]
Recently https://github.com/influxdata/influxdb/pull/25312 was merged in to 1.x's master branch. This is upcoming work that should be part of our v1.12 release version.

**Comment by bcutter** [comment]
Great news! How can we track if/when v1.12 is released? The release section of this repository seems to only provide series 2 releases.

**Comment by bcutter** [comment]
Unfortunately I couldn't figure out which release https://github.com/hassio-addons/addon-influxdb/releases is running currently.

In the InfluxDB itself in the configuration section it reports "Chronograf Version: 1.10.2". Is that indicator of help to determine the version?

**Comment by davidby-influx** [comment]
InfluxDB will print the version number at the beginning of the log on start-up.

We have not released the versions which contain the fix mentioned above, though community members may build them from source if they want pre-release versions.

**Comment by bcutter** [comment]
Wait - it's been 6 months. When will there be an actual release containing the change? 2025?

**Comment by davidby-influx** [comment]
Definitely in 2025.

**Comment by bcutter** [comment]
OK thx.

Take your time. This FR is only 8 years old, a few months don't matter.

Looking forward.
